File: app/app-learncode/src/main/java/com/open/learncode/设计模式/单利模式/静态内部类/Singlnton.java
package com.open.learncode.设计模式.单利模式.静态内部类;

public class Singlnton {

    public Singlnton() {

    }

    private static class SinglntonHolder {
        public static Singlnton INSTANCE = new Singlnton();
    }

    private static Singlnton getInstance() {
        return SinglntonHolder.INSTANCE;
    }
}

File: app/src/main/java/com/open/learncode/设计模式/单利模式/Kotlin中的单例/Singlnton.kt
package com.open.learncode.设计模式.单利模式.Kotlin中的单例

class Singlnton {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {

        }

        val instance: Singlnton by lazy { Singlnton() }
    }
}